identity/src/main/java/com/android/identity/flow/environment/Storage.kt

```diff
@@ -0,0 +1,55 @@
+package com.android.identity.flow.environment
+
+import kotlinx.io.bytestring.ByteString
+
+/**
+ * Simple persistent storage interface.
+ *
+ * Data records are organized using tables, keys, and peerIds. Table must be an ASCII string
+ * that does not come from an external source (as it is not escaped). It may or may not be
+ * case-sensitive. Key is a string that uniquely identifies a record in a table. Additionally,
+ * peerId is a string that identifies a "counterpart" entity, i.e. a particular client in the
+ * server environment, or a particular server in the client environment. In most cases all
+ * three record identifiers must be provided (table, peerId, and the key).
+ *
+ * Payload of the record is always just a blob of data. Storage does not interpret that data
+ * in any way.
+ */
+interface Storage {
```

Comment:
Some questions:

- Why can't we just use existing `StorageEngine` interface?
- Are we intending to use this interface only on the wallet-server side? Or in WSMs? Or both?
- What kind of implementations do we expect? One for our developer-grade Utopia Registry (backed by files?) and potentially a production-grade environment (backed by a database?)

Reply:
It is mostly for the server, so existing StorageEngine would not cut it - besides I would need to do hacky dance with joining table, peerId and key into a single key. 

There are two implementations, one on the client in the wallet app (in "remote" package, only used to run server code on the client for development) and the other on the server in wallet-server module, that one is for real. Both use database (SQLite on the client and JDBC interface on the server). For server dev environment I use pure java SQL implementation (added as a dependency, so no need for a developer to install or set up anything), but one can set up a real database (e.g MySQL) and point to it through a setting (did not test this code yet, but I've used this setup many times before).